### PR TITLE
adding support for golangci linting and formatting

### PR DESCRIPTION
**Pull request by shubhamavi**
This PR adds github actions for pointing out linting issues in PRs pushed for master or release- branches
As part of adding the action, this PR addresses all the existing issues pointed out by the linter.
The action should get active once the PR is merged, for future PRs.

An exhaustive list of supported linters can be found here: https://golangci-lint.run/usage/linters/ 
The proposed configuration (see .golangci.yml) only attaches basic linters without changing too much in upstream.

The PR fixes multiple misspells, lock copying, deadcode, unused vars etc.

### CODE REVIEW

File: tests/scaletest/scale_test.go [hunk 1]
```diff
@@ -199,15 +205,14 @@ func RebootAko(t *testing.T, wg *sync.WaitGroup) {
...
-		os.Exit(0)
+		ExitWithErrorf(t, "Cannot reboot Ako pod as : %v", err)
 	}
 	t.Logf("Ako rebooted.")
 	defer wg.Done()
 }
 
 /* Reboots Controller/Node/Ako if Reboot is set to true */
-func CheckReboot(t *testing.T, wg *sync.WaitGroup) {
+func (t *testing.T, wg *sync.WaitGroup) {
```
Comment (akshayhavile) [inline tests/scaletest/scale_test.go:215]: We are missing function name here.

Comment (shubhamavi) [reply tests/scaletest/scale_test.go:215]: thanks, i'll check this out, it's still in progress though, will add reviewers later

Comment (shubhamavi) [reply tests/scaletest/scale_test.go]: Fixed

**Comment by shubhamavi** [comment]
build ako

**Review by sudswasavi — APPROVED** [review]

**Review by monotosh-avi — APPROVED** [review]

**Pull request merged**